CesiumGltf/src/Model.cpp

```diff
@@ -208,3 +211,366 @@ void Model::merge(Model&& rhs) {
     this->scene = int32_t(this->scenes.size() - 1);
   }
 }
+
+namespace {
+void forEachPrimitiveInMeshObject(
+    const glm::dmat4x4& transform,
+    const Model& model,
+    const Node& node,
+    const Mesh& mesh,
+    std::function<Model::ForEachPrimitiveInSceneConstCallback>& callback) {
```

Comment:
nitpick from me but I wonder if replacing std::function with a template maybe a little faster than std::function. Std::function may allocate if the lambda is large enough

Reply:
cc@kring for more opinion

Reply:
Yeah I agree, this would be better as a template. There's no good reason I wrote it as a std::function originally.

Reply:
@baothientran Just so I understand, should the template callback type only be used for the local static functions, or should it be used for the public `forEachPrimitiveInScene` functions as well?